src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -1576,6 +1638,32 @@ public virtual ConcurrentDictionary<string, PackageResult> Upgrade(ChocolateyCon
                             {
                                 if (packageToUninstall != null)
                                 {
+                                    // Are we attempting a downgrade? We need to ensure it's allowed...
+                                    if (!config.AllowDowngrade && packageToUninstall.Identity.HasVersion && packageDependencyInfo.HasVersion && packageDependencyInfo.Version < packageToUninstall.Identity.Version)
+                                    {
+                                        var logMessage = StringResources.ErrorMessages.UnableToDowngrade.FormatWith(packageToUninstall.Name, packageToUninstall.Version, Environment.NewLine);
+                                        var nullResult = packageResultsToReturn.GetOrAdd(packageToUninstall.Name, packageToUninstall);
+                                        nullResult.Messages.Add(new ResultMessage(ResultType.Error, logMessage));
+                                        this.Log().Error(ChocolateyLoggers.Important, logMessage);
+                                        
+                                        if (config.Features.StopOnFirstPackageFailure)
+                                        {
+                                            throw new ApplicationException("Stopping further execution as {0} has failed.".FormatWith(packageDependencyInfo.Id));
+                                        }
+
+                                        continue;
+                                    }
+
+                                    // Package was previously marked inconclusive (not upgraded). We need remove the message since we are now upgrading it.
+                                    // Packages that are inconclusive but successfull are not labeled as successful at the end of the run.

```

Comment:
```suggestion
                                    // Packages that are inconclusive but successful are not labeled as successful at the end of the run.

```

Reply:
I don't see the problem here... (I've corrected it, but not pushed up yet.)